installing/disconnected_install/installing-mirroring-disconnected-v2.adoc

```diff
@@ -0,0 +1,215 @@
+
+This has its pros and cons, but due to the large amount of bugs been filed, we decided to change the behaviour in v2.
+
+In v2 a new kind called "DeleteImageSetConfiguration" has been introduced, this is used to configure the deletion of images, to avoid using an ImageSetConfig, accidentally deleting wanted (deployed) images.
+
+A new oc-mirror delete sub command is now available to execute the delete workflow with the DeleteImageSetConfiguration input.
+
+This functionality is able to work in a complete "air-gapped" (disconnected) environment, provided that the images have been previously mirrored (i.e the local cache and remote registry know about the images being deleted).
+
+The delete functionality is split into 2 stages:
+
+- The first stage is to create (generate) a delete yaml file using the --generate flag, this file is used to validate the images/blobs that will be deleted.
+- The second stage is to use the delete yaml file (once validated) to actually perform a local cache and remote registry delete.
+
+.For stage 1:
+** Use the following command to create `delete` yaml file for delete phase:
+
+[source,terminal]
+----
+$ oc-mirror delete --config delete-image-set-config.yaml --workspace file://<previously-mirrored-work-folder> --v2 --generate docker://<remote-registry> 
+----
+
+** Use the following command to compare versions:
+
+[source,terminal]
+----
+$ oc-mirror delete --config delete-image-set-config.yaml --workspace file://<previously-mirrored-work-folder> --v2 --generate --delete-id v4.11 docker://<remote-registry>
+----
+
+.For stage 2:
+** Once the generate stage is completed, use the following command to delete the images from the remote registry. 
+
+[source,terminal]
+----
+$ oc-mirror delete --v2 --delete-yaml-file <previously-mirrored-work-folder>/delete/delete-images-v4.11.yaml docker://<remote-registry>  
+----
+
+** Use the following command to delete the remote registry and forces the cache delete:
+
+[source,terminal]
+----
+$ oc-mirror delete --v2 --delete-yaml-file <previously-mirrored-work-folder>/delete/delete-images-v4.11.yaml --force-cache-delete true docker://<remote-registry> 
+----
+
+Here is an example of the `DeleteImageSetConfig`:
+
+[source,yaml]
+----
+apiVersion: mirror.openshift.io/v1alpha2
+kind: DeleteImageSetConfiguration
+delete:
+  platform:
+    channels:
+      - name: stable-4.13 
+        minVersion: 4.13.3
+        maxVersion: 4.13.3
+  operators:
+    - catalog: registry.redhat.io/redhat/redhat-operator-index:v4.12
+      packages:
+      - name: aws-load-balancer-operator
+  additionalImages: 
+    - name: registry.redhat.io/ubi8/ubi@sha256:bce7e9f69fb7d4533447232478fd825811c760288f87a35699f9c8f030f2c1a6
+    - name: registry.redhat.io/ubi8/ubi-minimal@sha256:8bedbe742f140108897fb3532068e8316900d9814f399d676ac78b46e740e34e
+----
```

Comment:
With this deleteImageSetConfig, oc-mirror will remove:
* all images of OCP release 4.13.3, 
* the catalog image redhat-operator-index v4.12
* the latest aws-load-balancer-operator bundle and all its related images
* as well as the additional images ubi and ubi-minimal referenced by their corresponding digests.

BTW, for this example, `aws-load-balancer-operator` it is best to specify a min and max version here, otherwise, it might not even find the operator to be deleted...

Reply:
Understood.